src/radical/pilot/agent/agent_0.py

```diff
@@ -636,32 +641,49 @@ def _ctrl_cancel_pilots(self, msg):
 
     # --------------------------------------------------------------------------
     #
-    def _ctrl_service_up(self, msg):
+    def _ctrl_service_info(self, msg, arg):
 
-        uid = msg['arg']['uid']
+        uid   = arg['uid']
+        name  = arg['name']
+        info  = arg['info']
+        ranks = arg['ranks']
+        rank  = arg['rank']
 
         # This message signals that an agent service instance is up and running.
         # We expect to find the service UID in args and can then unblock the
         # service startup wait for that uid
 
-        if uid not in self._service_uids_launched:
+        if uid != self._service_uid_launched:
             # we do not know this service instance
-            self._log.warn('ignore service startup signal for %s', uid)
+            self._log.warn('ignore service startup signal for %s [%s]', uid,
+                           self._service_uid_launched)
             return True
 
         if uid in self._service_uids_running:
             self._log.warn('duplicated service startup signal for %s', uid)
             return True
 
-        self._service_uids_running.append(uid)
-        self._log.debug('service %s started (%s / %s)', uid,
-                        len(self._service_uids_running),
-                        len(self._service_uids_launched))
+        # collect all rank info before we consider sucess
+        service_info = self._service_infos[name]
```

Comment:
sorry, seems like I'm missing, but where `self._service_infos` is filled up?

btw, in L407, we do `self._service_infos[tid] = dict()` - using `tid` and not `name`

Reply:
Dammit, what happened here?  Maybe I screwed up some devel-merge.  I'll recover that code...

Reply:
@mtitov : I cleaned up the service info handling, and specifically changed that so only rank 0 provides that info.  Otherwise the code was too complex for no noticeable benefit ...